server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -651,10 +652,13 @@ public void run() {
         err.set(null);
         JOB_HOLDER.reset();
 
-        // Attempt to process all existing log sorting work serially in this thread.
-        // When no work remains, this call will return so that we can look for compaction
-        // work.
-        logSorter.sortLogsIfNeeded();
+        if (System.nanoTime() > nextSortLogsCheckTime) {
```

Comment:
Usually need  to subtract two nano times to compute a duration when using nano time. Iits not always safe directly compare nano times because it can wrap, going from something like Long.MAX-10 to Long.MAX-10+20 which would go from positive to negative.  I am not actually sure if there is a problem here, just know this does not follow the usual pattern of using nano time to compute a duration. When computing durations those should always be positive.

Reply:
I changed this to millis to avoid doing something incorrect.